### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by kw2542**
Fix broken link to JoinExample.

### CODE REVIEW

**Comment by prateekm** [comment]
@PawasChhokra Can you take a look.

> Add link to samza-container.html on programming-model.html
@kw2542 It's not obvious why container documentation relevant for the programming model. Also, is that part of the new documentation or the old one?

**Comment by kw2542** [comment]
samza-container.html is missing entry point on the new documentation, I was guided by @rmatharu to add its link on programming-model.html.

@rmatharu could you help double check where samza-container.html fits most?

**Comment by rmatharu** [comment]
Yes this one is fine, earlier we used to have 
https://samza.apache.org/learn/documentation/0.14/api/overview.html link to samza-container page, but since that is not there anymore, the one you added should be fine.

File: docs/learn/documentation/versioned/container/state-management.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ One of the more interesting features of Samza is stateful stream processing. Tas
...
 
-However, being able to maintain state opens up many possibilities for sophisticated stream processing jobs: joining input streams, grouping messages and aggregating groups of messages. By analogy to SQL, the *select* and *where* clauses of a query are usually stateless, but *join*, *group by* and aggregation functions like *sum* and *count* require state. Samza doesn't yet provide a higher-level SQL-like language, but it does provide lower-level primitives that you can use to implement streaming aggregation and joins.
+However, being able to maintain state opens up many possibilities for sophisticated stream processing jobs: joining input streams, grouping messages and aggregating groups of messages. By analogy to SQL, the *select* and *where* clauses of a query are usually stateless, but *join*, *group by* and aggregation functions like *sum* and *count* require state. Samza supports [Samza SQL](/learn/documentation/{{site.version}}/api/samza-sql.html) for higher-level SQL-like language and [Beam API](/learn/documentation/{{site.version}}/api/beam-api.html) for streaming aggregation and joins.
```
Comment (prateekm) [inline docs/learn/documentation/versioned/container/state-management.md]: @kw2542 Is this page reachable from the existing documentation? Or is this part of the old documentation as well?

Comment (kw2542) [reply docs/learn/documentation/versioned/container/state-management.md]: Nice catch, I believe this is part of the old documentation, I will revert this change.

**Pull request merged**